Docstring controller part1

Helper2020 opened the pull request:
## Overview

1. This PR fixes part of [#11776].
2. This PR does the following: Added/updated args, return and raises in docstrings

## Essential Checklist

- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...",
followed by a short, clear summary of the changes.
- [x] I have followed the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- [x] The linter/Karma presubmit checks have passed on my local machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)

Helper2020 committed:
subtopic_viewer.py removed unaccessed variable in get method

Helper2020 committed:
Modified or added docstrings to functions in 'suggestion.py'

Helper2020 committed:
Modifed or added function docstrings in suggestion.py

Helper2020 committed:
fix liniing issue in line 352

Helper2020 committed:
Modified docstring function

Helper2020 committed:
Addressed linter check failure

Helper2020 committed:
Addressed indentation and trailing whitespace issue

Helper2020 committed:
Addressed trailing whitespace issue

Helper2020 committed:
Addressed trailing whitespace issue and indentation

Helper2020 committed:
Addressed linting issue w9024

Helper2020 committed:
Addressed linting issue again

Helper2020 committed:
Attempt to fix linting issue line too long

Helper2020 committed:
Attempted to address whitespace issue

Helper2020 committed:
Fixed linting errors

Helper2020 committed:
Fixed trailing whitespace

Helper2020 committed:
Fixed trailing whitespace

Helper2020 committed:
Added docstrings

Helper2020 committed:
Fix linting errors

Helper2020 committed:
fixed 8 space indentation

Helper2020 committed:
Attempting to address linting errors

Helper2020 committed:
Shorted raises in docstroing

Helper2020 committed:
Added args to docstring get function

Helper2020 committed:
modifed docstrings to some functions in  editor.py

Helper2020 committed:
Modified or added docstrings to controller.editor.py

Helper2020 committed:
Added and edited docstrings. in controller/contributor_dashboard.py

Helper2020 committed:
Finished writing and editing docstrings in controller/editor.py

Helper2020 committed:
Edited docstrings in contributer_dashboard.py

Helper2020 committed:
Added or edited docstrings in story_viewer.py

Helper2020 commented:
Hello it seems like some of the files docstrings were updated already before my  pr request. Need advice on how I should resolve the conflicts. Moreover in the conflicted files , I noticed the docstrings that contained sentences such as "Handles GET requests" were removed then rewritten in a more descriptive matter. Should I have done that for all the get/post requests docstrings?

seanlip commented:
@Helper2020 I think if the existing docstrings look good, then you can keep them, otherwise replace them with the better ones.

For the handler docstrings, I would suggest doing whatever is more useful for future developers. "Handles X requests" is probably OK, but more elaborate docstrings that provide context would be better.

lkbhitesh07 committed:
Adds a test that verifies the platform param appears once in the features or params list (#18355)

* initial changes

* added the feature handler at backend and made the changes in the frontend

* fixed lint

* intial changes for e2e test file for release-coordinator

* niy

* introduced webdriverio file for release coordinator and shifted the code from AdminPage js

* fixed lint

* nit change in e2e test

* nits to fix e2e tests

* nit to fix the e2e test

* corrected the hover-over-name of features tab

* initial changes

* created constant for the url and tried adding loading screen

* nits

* fix for e2e test

* nit fix for the e2e test

* added the platform param tab

* fixed lint

* fixed more lint

* tried adding loading screen

* fixed the frontend test

* fixed lint

* nit

* initial changes to migrate promo-bar-enabled config property

* added the loading screen on the features tab

* added the parameters-tab component to release coordinator page

* migrated the promo-bar config property

* done suggested changes

* fixed frontend tests

* fixed lint

* updated the promo bar on the release corrdinator page

* added the loading screen for the platform params tab

* fixed lint

* fixed mypy

* renamed the loadingScreenIsShown variable

* fixed backend tests

* fixed lint

* fixed backend coverage

* nit

* added the test

* fixed tests

* fixed tests

* fixed tests

* done suggested changes

* fixed tests

* fixed lint

* fixed mypy

* nit

* fixed backend tests

* fixed lint and typecheck error

* fixed more test

* nit

* done suggested changes

Helper2020 committed:
Merge branch 'develop' into docstring-controller-part1

Ash-2k3 committed:
GSoC_M1.3 : Add translation Length label (#18392)

* Add translation Length label

* Update core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

Co-authored-by: Sagang Wee <9004726+sagangwee@users.noreply.github.com>

* Update core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

Co-authored-by: Sagang Wee <9004726+sagangwee@users.noreply.github.com>

* Update core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

Co-authored-by: Sagang Wee <9004726+sagangwee@users.noreply.github.com>

* Address review comments

* Fix linting

* Update tessts

* Fix variable name

* Fix variable name

* Fix frontend tests

* Fix frontend tests

* Address review comments

* Update core/templates/pages/contributor-dashboard-page/opportunities-list-item/opportunities-list-item.component.html

Co-authored-by: Sagang Wee <9004726+sagangwee@users.noreply.github.com>

* Add frontend tests for label

* Fix linting

* Fix failing tests

* Fix failing tests

* Fix failing tests

* Only short labels will be shown to reviewers

* Fix frontend tests

* Add properties to suggesion object for acceptance tests

* Address PR comments

* Fix linting

* Fix failing acceptance tests

* Fix frontend tests

* Fix frontend tests

* Update opportunities-list-item.component.html

* Update opportunities-list-item.component.spec.ts

---------

Co-authored-by: Sagang Wee <9004726+sagangwee@users.noreply.github.com>

lkbhitesh07 committed:
Migrate some of the config properties to feature-flags/platform-params (#18354)

* initial changes

* added the feature handler at backend and made the changes in the frontend

* fixed lint

* intial changes for e2e test file for release-coordinator

* niy

* introduced webdriverio file for release coordinator and shifted the code from AdminPage js

* fixed lint

* nit change in e2e test

* nits to fix e2e tests

* nit to fix the e2e test

* corrected the hover-over-name of features tab

* initial changes

* created constant for the url and tried adding loading screen

* nits

* fix for e2e test

* nit fix for the e2e test

* added the platform param tab

* fixed lint

* fixed more lint

* tried adding loading screen

* fixed the frontend test

* fixed lint

* nit

* initial changes to migrate promo-bar-enabled config property

* added the loading screen on the features tab

* added the parameters-tab component to release coordinator page

* migrated the promo-bar config property

* done suggested changes

* fixed frontend tests

* fixed lint

* updated the promo bar on the release corrdinator page

* added the loading screen for the platform params tab

* fixed lint

* fixed mypy

* renamed the loadingScreenIsShown variable

* fixed backend tests

* fixed lint

* fixed backend coverage

* nit

* migrated improvements config property to feature flag

* changes to fix e2e tests

* nit

* fixed tests

* migrated learners-group config to feature flag

* fixed e2e test

* done suggested changes

* fixed lint and backend tests

* fixed e2e

* nit

* removed the unneeded assert check

* migrated always-ask-for-answer-details to platform param

* nit fix for tests

* migrated high_bounce_rate_task_state_bounce_rate_creation_threshold

* migrated high bounce minimum-exploration and rate-obsoletion

* nits

* nit

* rearranged the params in test

* sorted the params names in test

* nit

* fixed failing tests

---------

Co-authored-by: lkbhitesh07 <lkbhiteh07@gmail.com>

Helper2020 committed:
Fixed linting errors in editor.pu

Helper2020 committed:
fixed linting errors topic_and_skills_dashboard.py

Helper2020 commented:
Hello , from looking at the error logs I see the error

ERROR:root:Exception raised at http://localhost/create/0: get() takes 1 positional argument but 2 were given
Traceback (most recent call last):
  File "/home/runner/work/oppia/oppia/third_party/python_libs/webapp2.py", line 604, in dispatch
    return method(*args, **kwargs)
  File "/home/runner/work/oppia/oppia/core/controllers/acl_decorators.py", line 254, in test_can_play
    return handler(self, exploration_id, **kwargs)
TypeError: get() takes 1 positional argument but 2 were given
ERROR:root:Exception raised: get() takes 1 positional argument but 2 were given

I am not sure how to solve this. I don't recall modifying the acl_decorators.py. Not sure what to do.

Helper2020 committed:
Merge branch 'develop' into docstring-controller-part1

lkbhitesh07 committed:
Edit default value for feature flag and platform parameter (#18413)

* initial changes

* fixed tests

* fixed e2e test

* fixed lint

* fixed the failing tests

* ts check fix

* fixed frontend tests

* updates to the features tab

* bit

* changed feature stage of dummy-feature

* fixed backend test

* added blank line to the spec file

* change the default value of feature flag

* added the default_value field to the model

* added for feature flags

* handled the incoming schema for feature flags

* fixed backend tests

* fixed tests

* fixed backend tests

* fixed tests

* fixed lint

* fixed single mypy

* fixed mypy tests

* more mypy

* added button to update the default values

* fixed lint

* more lint

* nit

* return default value from parameter registry when not present in storage model

* added test

* fixed lint

* fixed mypy

---------

Co-authored-by: lkbhitesh07 <lkbhiteh07@gmail.com>

Ykumar1415 committed:
Fix #18333 [Feature request]:Fixed The Problem Of Limited Text Formatting #18333  (#18397)

* limited text formatting error fixed

* Update blog-dashboard-page.component.html

* Update goals-tab.component.ts

* Update goals-tab.component.ts

gupta-arpan committed:
Fix #18540: Update modal labels to Publish Changes (#18570)

Update modal labels to Publish Changes

Co-authored-by: Arpan Gupta <gupta-arpan@pop-os.localdomain>

seanlip committed:
Complete some miscellaneous small fixes -- drop UA tag, add comments and update issue templates. (#18553)

* Misc small fixes -- update Android page, add comments and update issue templates, drop UA tag.

* Drop changes for updating Android page description.

Patel-Muhammad committed:
Contributor dashboard -- Fixes axeDev tools + lighthouse error and makes the lighthouse score to 1 (#18454)

added a11y fixes

Patel-Muhammad committed:
Added i18n for user facing texts in the question player and story revision page (#18491)

* added i18n

* added i18n in qqq.json

* fixed tests

prafulbbandre committed:
Fix #18153: Submit button text is visible properly. (#18485)

* fixed state of submit button

* fixed state of submit button

* fixed lint

thesatyam05 committed:
issue #17997 Fixed Misaligned Text in blog-home-page.component.html ( synced pull ) (#18547)

Updated branch for pull req : #18472 and issue: #17997

Fixed the misalignment of the text "Sorry, there are no blog posts to show." in file blog-home-page.component.html

Adarsh7825 committed:
Fix auto-hover issue in Community Lessons section (#18451)

* make changes in oppia.css file

* Fixed the auto-hover

---------

Co-authored-by: Kshitij Patil <91470808+kshitij01042002@users.noreply.github.com>

Kej-r03 committed:
Fix part of #16535: Modify chapter editor page: GSoC M: 1.5 (#18455)

* Modify chapter editor page

* Modify chapter editor UI

* Rectify PR

* Add constant

* Add local constant

SubhashThenua committed:
Add feature flag for redesigned learner dashboard (#18569)

* added feature flag for redesigned learner dashboard

* changes flag to DEV

---------

Co-authored-by: JSRKJSRK <SubhashThenua>

gupta-arpan committed:
Fix issue #18555: Ensure skills tab visibility on mobiles (#18574)

Ensure Skills tab visibility on mobile devices

Co-authored-by: Arpan Gupta <gupta-arpan@pop-os.localdomain>

lkbhitesh07 committed:
Remove some of the config properties (#18306)

* removed checkpoints config property

* lints

* fixed e2e test

* fixed another e2e test

* removed classroom-promos config property

* fixed lint

* more lints

* removed contributor-dashboard config property

* removed always-ask-for-answer-details config property

* lint

* removed batch-index-for-mailchimp config property

* fixed backend test

* fixed frontend tests

* removed max-number-of-svgs-math config property

* removed max-exps-math-svgs

* Updated the list of depricated properties

* fixed lint

* fixed frontend tests

* restored the always-ask config property and did suggested changes

* nit

* error raises in case of bad backend request

* fixed lint

* added contributor_dashboard_is_enabled to the depricated list

* removed the previously added error catch blocks

* fixed lints

* fixed lint

* nit

* fixed lighthouse checks

* try to fix e2e test

* try to fix the e2e

* lint fix

* try to fix e2e

* fixed the e2e test

* fixed the e2e test

* nit

* nit fix

---------

Co-authored-by: lkbhitesh07 <lkbhiteh07@gmail.com>

Commit:
Revert "Beam Job for populating new models used in Contributor Admin Dashboard: GSoC M1.4" (#18568)

Revert "Beam Job for populating new models used in Contributor Admin Dashboard: GSoC M1.4 (#18406)"

This reverts commit bceb05fd6830147bf8292638f7a534c3db36e05e.

Patel-Muhammad committed:
Home page -- Fixes axeDev tools + lighthouse error and makes the lighthouse score to 1 (#18563)

added tabindices

Helper2020 committed:
Merge remote-tracking branch 'upstream/develop' into docstring-controller-part1

Helper2020 commented:
@seanlip Hello Sean
So here was the function before I removed unused argument from controllers.editor.py
`
class TopUnresolvedAnswersHandler(
    base.BaseHandler[Dict[str, str], Dict[str, str]]
):
    """Returns a list of top N unresolved answers."""

    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
    URL_PATH_ARGS_SCHEMAS = {
        'exploration_id': {
            'schema': SCHEMA_FOR_EXPLORATION_ID
        }
    }
    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}

    @acl_decorators.can_edit_exploration
    def get(self, unused_exploration_id: str) -> None:
        """Handles GET requests for unresolved answers."""
        # TODO(#11475): Return visualizations info based on Apache Beam job.
        self.render_json({'unresolved_answers': []})`

I removed the unsed_exploration_id. It seems it's used as a place holder?
Here is the test (I think) from test_test.py.
`
class TopUnresolvedAnswersHandlerTests(test_utils.GenericTestBase):

    def setUp(self) -> None:
        super().setUp()
        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)

        self.exp_id = 'exp_id'
        self.exploration = self.save_new_valid_exploration(
            self.exp_id, self.owner_id)

    def test_cannot_get_unresolved_answers_with_no_state_name(self) -> None:
        self.login(self.OWNER_EMAIL)

        response = self.get_json(
            '/createhandler/get_top_unresolved_answers/%s' % self.exp_id,
            expected_status_int=200)

        self.assertEqual(response['unresolved_answers'], [])

        self.logout()

`
The error after running backend test
<img width="1087" alt="image" src="https://github.com/oppia/oppia/assets/12247834/7c9b3419-b1e1-42e0-8b02-e260619bcc9b">

How should I proceed?

seanlip commented:
@Helper2020 You're modifying a decorator here, so I don't think you can remove the unused_exploration_id argument, because that argument is also being used in the method(s) that that decorator is decorating.

I think you will see more comprehensive details on the backend error if you scroll up through the logs a bit more. Usually a 500 error comes with a stack trace, which should help you with debugging.

Helper2020 committed:
Fixed functions that failed backend tests

Helper2020 committed:
Fixed backend test error for subtopic_viewer.py
